rln/src/protocol.rs

```diff
@@ -103,121 +171,90 @@ pub enum ProofError {
     SynthesisError(#[from] SynthesisError),
 }
 
-// XXX This is different from zk-kit API:
-// const witness = RLN.genWitness(secretHash, merkleProof, epoch, signal, rlnIdentifier)
-// const fullProof = await RLN.genProof(witness, wasmFilePath, finalZkeyPath)
-//
-
-// TODO Change API here
-/// Generates a semaphore proof
+/// Generates a RLN proof
 ///
 /// # Errors
 ///
 /// Returns a [`ProofError`] if proving fails.
 pub fn generate_proof(
-    identity: &Identity,
-    merkle_proof: &merkle_tree::Proof<PoseidonHash>,
-    external_nullifier_hash: Field,
-    signal_hash: Field,
-) -> Result<Proof, ProofError> {
-    // TODO Fix inputs
-    // Semaphore genWitness corresponds to these
-    // RLN different, should be:
-    // identity_secret
-    // path_elements (merkleProof.siblings))
-    // identity_path_index (merkleProof.pathIndices)
-    // x (RLN.genSignalHash(signal), assuming shouldHash is true)
-    // epoch
-    // rln_identifier
-    let inputs = [
-        // FIXME should be identity_secret, not just nullifier!
-        ("identity_secret", vec![identity.nullifier]),
-        //("identityTrapdoor", vec![identity.trapdoor]),
-        ("path_elements", merkle_proof_to_vec(merkle_proof)),
-        ("identity_path_index", merkle_proof.path_index()),
-        ("externalNullifier", vec![external_nullifier_hash]),
-        // XXX: Assuming signal is hashed
-        ("x", vec![signal_hash]),
-        // FIXME epoch just hardcoded to random value
-        ("epoch", vec![signal_hash]),
-        // FIXME rln_identifier just hardcoded to random value
-        ("rln_identifier", vec![signal_hash]),
-    ];
-    let inputs = inputs.into_iter().map(|(name, values)| {
-        (
-            name.to_string(),
-            values.iter().copied().map(Into::into).collect::<Vec<_>>(),
-        )
-    });
-
+    mut builder: CircomBuilder<Bn254>,
+    proving_key: &ProvingKey<Bn254>,
+    rln_witness: RLNWitnessInput,
+) -> Result<(Proof, Vec<Fr>), ProofError> {
```

Comment:
What is in Vec<Fr> output? inputs? aka public outputs? Can be clarified

Reply:
Now this is removed, since users should not compute public circuit's input/output values evaluating witnesses in circuit. Utility functions will now do it directly from witnesses and generate_proof now only returns a zkSNARK proof.